systems/controllers/constrained_lqr_controller.cc

```diff
@@ -0,0 +1,141 @@
+#include "systems/controllers/constrained_lqr_controller.h"
+
+namespace dairlib {
+namespace systems {
+
+using std::make_unique;
+using std::unique_ptr;
+using Eigen::VectorXd;
+using Eigen::MatrixXd;
+using Eigen::HouseholderQR;
+using dairlib::multibody::ContactInfo;
+using dairlib::multibody::ContactToolkit;
+using drake::AutoDiffXd;
+using drake::AutoDiffVecXd;
+using drake::math::initializeAutoDiff;
+using drake::math::autoDiffToGradientMatrix;
+using drake::math::autoDiffToValueMatrix;
+using drake::systems::Context;
+using drake::systems::controllers::LinearQuadraticRegulator;
+using drake::systems::controllers::LinearQuadraticRegulatorResult;
+
+ConstrainedLQRController::ConstrainedLQRController(
+    const RigidBodyTree<double>& tree, VectorXd q, VectorXd u, VectorXd lambda,
+    MatrixXd Q, MatrixXd R, ContactInfo contact_info)
+    : tree_(tree),
+      contact_info_(contact_info),
+      contact_toolkit_(
+          make_unique<ContactToolkit<AutoDiffXd>>(tree, contact_info)),
+      num_positions_(tree.get_num_positions()),
+      num_velocities_(tree.get_num_velocities()),
+      num_states_(num_positions_ + num_velocities_),
+      num_efforts_(tree.get_num_actuators()),
+      num_forces_(tree.getNumPositionConstraints() +
+                  contact_info.num_contacts * 3) {
+  // Input port that takes in an OutputVector containing the current Cassie
+  // state
+  input_port_info_index_ =
+      this
```

Comment:
Linebreak formatting looks off here

Reply:
For some reason my linting formats it this way. I'm using the Cland formatter.